
 kernel.org commits
 372a775f50347f5c1dd87752b16e5c05ea965790 [MIPS] Enable -ffunction-sections sections.
 0510617b85758b6e66f3c602ceccea1826440470 [MIPS] Fix data bus error recovery
 1a4ba061b3c93804027b4cc4c588cfad748d9fc2 MIPS: Use -mno-check-zero-division

---
 arch/mips/Makefile       |    3 ++-
 arch/mips/kernel/traps.c |    6 ++++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff -urNBp a/arch/mips/kernel/traps.c b/arch/mips/kernel/traps.c
--- a/arch/mips/kernel/traps.c	2011-02-22 03:00:00.000000000 +0300
+++ b/arch/mips/kernel/traps.c	2011-02-22 14:48:04.000000000 +0300
@@ -346,8 +345,8 @@ NORET_TYPE void ATTRIB_NORET die(const c
 	do_exit(SIGSEGV);
 }
 
-extern const struct exception_table_entry __start___dbe_table[];
-extern const struct exception_table_entry __stop___dbe_table[];
+extern struct exception_table_entry __start___dbe_table[];
+extern struct exception_table_entry __stop___dbe_table[];
 
 __asm__(
 "	.section	__dbe_table, \"a\"\n"
@@ -1578,4 +1577,6 @@ void __init trap_init(void)
 
 	flush_icache_range(ebase, ebase + 0x400);
 	flush_tlb_handlers();
+
+	sort_extable(__start___dbe_table, __stop___dbe_table);
 }
diff -urNBp a/arch/mips/Makefile b/arch/mips/Makefile
--- a/arch/mips/Makefile	2008-12-12 06:25:39.000000000 +0300
+++ b/arch/mips/Makefile	2011-02-22 14:48:04.000000000 +0300
@@ -12,7 +12,8 @@
 # for "archclean" cleaning up for this architecture.
 #
 
-cflags-y :=
+cflags-y := -ffunction-sections
+cflags-y += $(call cc-option, -mno-check-zero-division)
 
 #
 # Select the object file format to substitute into the linker script.
@@ -147,6 +147,8 @@
 cflags-$(CONFIG_CPU_R10000)	+= $(call cc-option,-march=r10000,-march=r8000) \
 			-Wa,--trap
 
+cflags-$(CONFIG_CPU_MIPS32)	+= -mno-branch-likely
+
 ifdef CONFIG_CPU_SB1
 ifdef CONFIG_SB1_PASS_1_WORKAROUNDS
 MODFLAGS	+= -msb1-pass1-workarounds
-- 
